packages/main/src/plugin/extension/extension-analyzer.spec.ts

```diff
@@ -0,0 +1,131 @@
+/**********************************************************************
+ * Copyright (C) 2023-2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import * as fs from 'node:fs';
+import { readFile, realpath } from 'node:fs/promises';
+import * as path from 'node:path';
+
+import { beforeEach, describe, expect, test, vi } from 'vitest';
+
+import { ExtensionAnalyzer } from './extension-analyzer.js';
+
+let extensionAnalyzer: ExtensionAnalyzer;
+
+vi.mock('node:fs');
+vi.mock('node:fs/promises');
+
+beforeEach(() => {
+  vi.resetAllMocks();
+  extensionAnalyzer = new ExtensionAnalyzer();
+});
+
+describe('analyze extension and main', async () => {
```

Comment:
```suggestion
describe('analyze extension and main', () => {
```

describe don't need to be async?

Reply:
removed, but it was a move of the existing content from extension-loader.spec.ts file